GLASGOW| MAY_2025 | YOUSEF ABDUALRHMAN | WIREFRAME

Yousefmohammed249 opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

-Moved the .hero section inside <main> for structural consistency
-Ensured the first article (README) spans the full width — aligning with the left edge of the second article and right edge of the third
-Applied flex box layout to the second and third articles using .flex-container
-Added CSS to left-align all child elements inside each article (text-align: left)
-Preserved semantic HTML structure and responsive layout
-Double check and fix all HTML code Errors.


## Questions

Ask any questions you have for your reviewer.

Yousefmohammed249 commented:
#CYF very things is god

cjyuan commented on Wireframe/index.html:
```diff
@@ -14,20 +14,44 @@ <h1>Wireframe</h1>
+        <h2>The Purpose of a Wireframe</h2>
+        <p>
+          A wireframe is a simple visual guide that shows the structure and layout of a webpage or app.
+           Its main purpose is to plan the design, organize content, and show how users will interact with the interface.
+           Wireframes help teams communicate ideas clearly, gather early feedback, and fix problems before moving to detailed design and development.
+        </p>
+        <a href="https://www.interaction-design.org/literature/topics/wireframe">Read more</a>
+      </article>
+       <article>
+        <img src="https://d8it4huxumps7.cloudfront.net/uploads/images/6536475adca0a_git_create_branch_01.jpg?d=2000x2000" alt="diagram explaining the purpose of branch in git" />
+        <h2>A branch in Git</h2>
+        <p>
+         A branch in Git is a separate version of a project used to develop features, fix bugs, or experiment without affecting the main code. 
```
The indentation of the code is a bit off.

You can use VSCode's "Format Document" feature to format your code for better readability and consistency.
To use the feature, right-click inside the code editor and select the option.
Please note that if there are syntax errors in the code, the "Prettier" extension may not format HTML code properly.

cjyuan reviewed (commented):
1. According to https://validator.w3.org/, there are errors and warnings in your code. Can you fix them?

2. When a wireframe is provided, our implementation should closely reflect its appearance and layout to ensure consistency with design expectations. You're off to a solid start. To better align with the wireframe, can you center the page title, the short description beneath the page title, and the text in the footer?

3. Can you format the checked boxes using the proper Markdown syntax in your PR description so that they look something like this?
![image](https://github.com/user-attachments/assets/6f85e6c5-4dd7-49aa-a9fc-e17a9dc2189d)
(With proper Markdown syntax, we can use mouse to check/uncheck the items)

4. In the PR description, can you also provide a brief description (under the "Changelist" section) summarizing the purpose of the PR and the changes you’ve made?
![image](https://github.com/user-attachments/assets/f510bc87-2e14-4e47-853b-65a7453d7b41)

cjyuan commented:
Please note that in CYF courses, the recommended way to inform the reviewer of your changes is to do both of the following:
  - **Reply to their feedback.**
    - In the responses, clarify how each piece of feedback was addressed to demonstrate that you've carefully reviewed the suggestions.
    - Your response may trigger a notification (depending on the reviewer's settings), helping ensure they’re aware of the updates you’ve made. 
  - **Replace the "Reviewed" label by a "Needs review" label** (which you have done -- great!)
    - Without this label, the reviewer would not know if your changes is ready to be reviewed.

cjyuan commented:
Your code is now error-free. Good job.

1. You still have not yet addressed this request:
> When a wireframe is provided, our implementation should closely reflect its appearance and layout to ensure consistency with design expectations. You're off to a solid start. To better align with the wireframe, **can you center the page title, the short description beneath the page title, and the text in the footer**?

2. You added the "Needs review" label but you didn't response to my comment. (This could slow down the review process because I don't get notified when you only change the label).

Yousefmohammed249 commented:
Great catch. I added full-height layout styling with flex-direction: column and made the main content expand, so now the footer stays fixed at the bottom of the viewport.

cjyuan commented:
You recent commit only involve changing `index.html. No change is done on CSS. The issues I pointed out in previous feedback still remains.

Yousefmohammed249 commented:
ok thank you for let me to know that

cjyuan commented:
Checkboxes in PR description fixed. Good job.

1. The footer is still not yet fixed at the bottom of the **viewport** (look up what viewport means).

2. Your recent change to the CSS messed up the webpage layout: https://deploy-preview-637--cyf-onboarding-module.netlify.app/wireframe/.

Yousefmohammed249 commented:
✅ What i Fixed:
The footer now stays at the bottom of the screen (position: fixed).

The first image fills its article width.

The second and third articles appear side-by-side on larger screens.

Page title and footer are centered.

cjyuan commented:
Changes look good. You made a lot of improvements.

In the wireframe, the left and right edges of the first article align with the left edge of the 2nd article and the right edge of the 3rd article. I believe you can easily implement that. 

I will go mark this PR as "Complete" first.

Yousefmohammed249 commented:
Wow 🤯 thank you so much 🙏

On Thu, Jun 5, 2025, 1:21 PM CJ Yuan ***@***.***> wrote:

> *cjyuan* left a comment (CodeYourFuture/Module-Onboarding#637)
> <https://github.com/CodeYourFuture/Module-Onboarding/pull/637#issuecomment-2944014617>
>
> Changes look good. You made a lot of improvements.
>
> In the wireframe, the left and right edges of the first article align with
> the left edge of the 2nd article and the right edge of the 3rd article. I
> believe you can easily implement that.
>
> I will go mark this PR as "Complete" first.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CodeYourFuture/Module-Onboarding/pull/637#issuecomment-2944014617>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BRWBNAQVAJ335TRR4PH2PW33CAY6LAVCNFSM6AAAAAB6BGR6G2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNBUGAYTINRRG4>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
